src/axl_pthread.c

```diff
@@ -0,0 +1,369 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <pthread.h>
+#include <sys/sysinfo.h>    /* get_nprocs() */
```

Comment:
can we use something other than sys/sysinfo? Mac's don't have that.

Reply:
Maybe?:
```
#include <unistd.h>
long number_of_processors = sysconf(_SC_NPROCESSORS_ONLN);
```
One commenter says it works on Linux and OSX:
https://stackoverflow.com/questions/4586405/how-to-get-the-number-of-cpus-in-linux-using-c/4586471#4586471